armi/reactor/blocks.py

```diff
@@ -910,10 +910,14 @@ def add(self, c):
 
         self.derivedMustUpdate = True
         self.clearCache()
-        mult = int(c.getDimension("mult"))
-        if self.p.percentBuByPin is None or len(self.p.percentBuByPin) < mult:
-            # this may be a little wasteful, but we can fix it later...
-            self.p.percentBuByPin = [0.0] * mult
+        try:
+            mult = int(c.getDimension("mult"))
```

Comment:
If one wanted to place a fuel compact full of TRISO on a fixed lattice (e.g., #224), it would be useful to leverage the `latticeIDs` feature to model the lattice. In that case, the compact should have a multiplicity reflecting the number of locations in this `Block`

Reply:
totally. I think if we just have a block that has a child called `triso compact` or something that happens to have matrix + kernels in it then it will look just like a 'pin' in non-triso fuel and should work with pin lattices all the same.